src/lib.rs

```diff
@@ -299,8 +351,9 @@ fn calculate_fee(weight: u32, rate: f32) -> u64 {
 }
 
 /// Returns the effective value which is the actual value minus the estimated fee of the OutputGroup
-fn effective_value(output: &OutputGroup, option: &CoinSelectionOpt) -> u64 {
-    unimplemented!()
+#[inline]
+fn effective_value(output: &OutputGroup, feerate: f32) -> u64 {
+    output.value - calculate_fee(output.weight, feerate)
```

Comment:
Make use of this [checked_sub](https://doc.rust-lang.org/std/primitive.u64.html#method.checked_sub) method and in case of None(underflow) return 0.